stmhal/usbd_hid_interface.c

```diff
@@ -45,22 +45,33 @@
 
 /* Private variables ---------------------------------------------------------*/
 
-static __IO uint8_t dev_is_connected = 0; // indicates if we are connected
-
 static uint8_t buffer[2][HID_DATA_FS_MAX_PACKET_SIZE]; // pair of buffers to read individual packets into
 static int8_t current_read_buffer = 0; // which buffer to read from
-static uint32_t last_read_len; // length of last read
+static uint32_t last_read_len = 0; // length of last read
 static int8_t current_write_buffer = 0; // which buffer to write to
 
-
 /* Private function prototypes -----------------------------------------------*/
+static int8_t HID_Itf_Init     (void);
 static int8_t HID_Itf_Receive  (uint8_t* pbuf, uint32_t Len);
 
 const USBD_HID_ItfTypeDef USBD_HID_fops = {
+    HID_Itf_Init,
     HID_Itf_Receive
 };
 
 /**
+  * @brief  HID_Itf_Init
+  *         Initializes the HID media low layer
+  * @param  None
+  * @retval Result of the opeartion: USBD_OK if all operations are OK else USBD_FAIL
+  */
+static int8_t HID_Itf_Init(void)
+{
+    USBD_HID_SetRxBuffer(&hUSBDDevice, buffer[current_write_buffer]);
```

Comment:
To make it robust, I think this function should initialise all the static variables (current_read_buffer, etc).